### DISCUSSION TITLE
Does Sphinx-Needs Support Expanding Needs In-Place?

### DISCUSSION THREAD

**Discussion by phathuynh97** [useblocks/sphinx-needs#1055] [General]
Hello,

I'm currently using Sphinx-Needs to generate my requirement traceability report, and I've encountered a challenge. When I click on a need, for example, DD_001, the content automatically jumps to another page where DD_001 is located. However, I'd prefer to stay on the same page and have an option, like a clickable arrow next to DD_001, that expands to display the content below that ID, as shown in the attached image.

[I can't attached the image here, so I use the text instead]
tags: flow_example
satisfies: [REQ_001](https://github.com/useblocks/sphinx-needs/01_requirement/00_requirement.html#REQ_001)
is satisfied by: [DD_001](https://github.com/useblocks/sphinx-needs/03_detail_design/00_detail_design.html#DD_001)
is tested by: [IT_001](https://github.com/useblocks/sphinx-needs/06_integration_test/00_integration_test.html#IT_001)

Could anyone confirm if Sphinx-Needs supports a feature like this?

Thank you in advance, and I hope you all have a great day!

**Comment by danwos** [comment]
Hi @phathuynh97,

for this use case Sphinx-Preview was created:
https://sphinx-preview.readthedocs.io/en/latest/

This supports your use case, if not please feel free to add more info about what is missing.

![sphinx-preview](https://sphinx-preview.readthedocs.io/en/latest/_images/sphinx-preview-showcase.gif)
